### DISCUSSION TITLE
Params are not validated?

### DISCUSSION THREAD

**Discussion by lockinmarv** [ardatan/feTS#896] [Q&A]
Hey!

When using a route like `/greetings/:id` with a required param `id`, the missing param isn't validated – neither on backend nor with the TS client. Am I missing something here and this is expected?

![CleanShot 2023-12-05 at 20 18 20](https://github.com/ardatan/feTS/assets/6840287/26b89160-fa3a-4af0-bdaf-317dc74b9156) 
![CleanShot 2023-12-05 at 20 20 27](https://github.com/ardatan/feTS/assets/6840287/837dd8f7-a0e7-4883-8c3e-b54a9b2c8089)

Within the Swagger UI it's required, though.

**Comment by ardatan** [comment]
You need to mark them required. Otherwise, it will throw an error only if the value is not a string.

**Reply by lockinmarv** [reply]
But the `Type.Object({ id: Type.String() })` should already mark the ID as required, no?

**Reply by ardatan** [reply]
Not really. You can also see it is not required in your `client`call. If it was required, it would be already required in your `client` call.

**Reply by lockinmarv** [reply]
How can I make it required then? The TypeBox documentation has this example which generates JSON schema with the required prop. 🤔 
![CleanShot 2023-12-05 at 22 19 42@2x](https://github.com/ardatan/feTS/assets/6840287/42eae827-2096-4aa1-9ea7-7d5c7a599d51)

As I said earlier, the Swagger UI marks it as required too.

![CleanShot 2023-12-05 at 22 26 58@2x](https://github.com/ardatan/feTS/assets/6840287/9fef1cc0-bcb1-4b1f-a64d-01234ea6ea6d)

This also doesn't work (copied from documentation)
![CleanShot 2023-12-05 at 22 29 30@2x](https://github.com/ardatan/feTS/assets/6840287/dea96157-5b74-4ef7-a813-f9edd3cd592c)

It works for query parameters without issues though, with just defining it as `Type.Object()`. It even throws a 400 Bad Request error.
